### ISSUE TITLE
UI Components: Fed sign in with Hosted UI bug

### ISSUE THREAD

**Issue by iartemiev** [aws-amplify/amplify-js#5232]
**Describe the bug**
Federated sign in using Hosted UI does not show user as logged in after getting redirected back to the app. Requires user to refresh browser twice before correct logged in state is shown. Reproduced with React and Angular 9.

**To Reproduce**

1. Configure federated sign in via Amplify CLI (issue occurs with Sign in with Amazon, fb, etc.)

2. Start app with the following code in App.js:
```jsx
<AmplifyAuthenticator federated={{}}>
  <div className="App">Main
  <AmplifySignOut />
  </div>
</AmplifyAuthenticator>
```

3. Use the **Sign in with AWS** button to sign in with Cognito Hosted UI

4. Use social provider button to sign in

5. Hosted UI redirects back to app

6. App still showing log in form. The following is logged to the console:

![image](https://user-images.githubusercontent.com/29709626/77949124-71e85800-7294-11ea-9ec6-8c8913277214.png)

7. Refreshing the page **twice** results in the app showing the user as logged in successfully.

**Expected behavior**

Expected behavior is getting signed in without having to refresh the page.

<details>
  <summary><strong>Environment</strong></summary>

```
npx envinfo --system --binaries --browsers --npmPackages --npmGlobalPackages

  System:
    OS: macOS 10.15.4
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 37.12 GB / 64.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.16.1 - /usr/local/bin/node
    Yarn: 1.22.4 - ~/.yarn/bin/yarn
    npm: 6.13.4 - /usr/local/bin/npm
  Browsers:
    Chrome: 80.0.3987.149
    Firefox: 74.0
    Safari: 13.1
  npmPackages:
    @aws-amplify/ui-react: ui-preview => 0.0.2-ui-preview.5545+6aae9f44 
    aws-amplify: ui-preview => 1.3.1-ui-preview.40+6aae9f44 
    react: ^16.8.6 => 16.13.1 
    react-dom: ^16.8.6 => 16.13.1 
    react-scripts: 2.1.8 => 2.1.8 
  npmGlobalPackages:
    @aws-amplify/cli: 4.17.2
    cordova: 9.0.0
    detox-cli: 10.0.7
    expo-cli: 3.15.5
    ionic: 5.4.16
    n: 6.3.0
    native-run: 0.3.0
    prettier: 1.19.1
    verdaccio: 4.4.3

```

</details>

**Comment by ericclemmons** [comment]
Here is the order of events I used:

1. http://localhost:3000/
1. Sign in with Google
1. Redirected back to http://localhost:3000/, **not** logged in.
1. Refreshed, which magically changed the URL to http://localhost:3000/?code=..., only showing **Loading...**
1. Refreshed again, which kept the URL **but successfully loaded the app!**

**tldr; `<amplify-authenticator>` isn't listening to `Hub` events, so the asynchronous `user` resolution doesn't change the state. It only works when it's already in Storage.**

Here are my findings:

- #5216 has been merged, but https://unpkg.com/browse/@aws-amplify/ui-components@0.1.1-ui-preview.40/dist/esm-es5/amplify-amazon-button_43.entry.js doesn't have the fix.  **The `ui-preview` build hasn't been updated since**, likely due to a build error.

- Refreshing a URL with `code` _will_ throw `invalid_grant` errors (the `code` has already been used). However, the `user` will exist in Storage and the app _should_ function.

- The component should be centered by default, but is not:

    > <img width="1315" alt="Screen Shot 2020-03-30 at 6 21 45 PM" src="https://user-images.githubusercontent.com/15182/77977026-62c4d280-72b3-11ea-81bb-3574903198bd.png">

- There should be an `– or –` between Federated + Email, but isn't.  (This is due to `amplify-sign-in` being unaware federated buttons will be rendered at all).

- `amplify-authenticator` isn't listening for Auth `Hub` events, so refreshing is the only way to trigger the current checks for `user`:

    https://github.com/aws-amplify/amplify-js/blob/38e55b090e10b71251439344365f6aa9344d049a/packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx#L32-L54

    Compare that to `aws-amplify-react`'s `Authenticator`, which listens to `Hub`:

    https://github.com/aws-amplify/amplify-js/blob/38e55b090e10b71251439344365f6aa9344d049a/packages/aws-amplify-react/src/Auth/Authenticator.tsx#L70-L79

    https://github.com/aws-amplify/amplify-js/blob/38e55b090e10b71251439344365f6aa9344d049a/packages/aws-amplify-react/src/Auth/Authenticator.tsx#L137-L160

**Comment by ericclemmons** [comment]
Fixed via https://github.com/aws-amplify/amplify-js/pull/5244
